lib/pymedphys/_interp/interp.py

```diff
@@ -0,0 +1,386 @@
+        )
+
+        diffs = np.zeros(3)
+        for i, axis in enumerate(axes_known):
+            diffs[i] = axis[1] - axis[0]
+
+        # pylint: disable=not-an-iterable
+        for i in nb.prange(points_interp.shape[0]):
+            xpi, ypi, zpi = (
+                points_interp[i, 0],
+                points_interp[i, 1],
+                points_interp[i, 2],
+            )
+
+            if (
+                not x[0] <= xpi <= x[-1]
+                or not y[0] <= ypi <= y[-1]
+                or not z[0] <= zpi <= z[-1]
+            ):
+                values_interp[i] = extrap_fill_value
+                continue
+
+            # Find the indices of the surrounding grid points
+            x1_idx = np.searchsorted(x, xpi)
+            x0_idx = x1_idx - 1
+            y1_idx = np.searchsorted(y, ypi)
+            y0_idx = y1_idx - 1
+            z1_idx = np.searchsorted(z, zpi)
+            z0_idx = z1_idx - 1
+
+            if x0_idx < 0:
+                x0_idx = 0
+            if y0_idx < 0:
+                y0_idx = 0
+            if z0_idx < 0:
+                z0_idx = 0
+            if x1_idx >= x.size:
+                x1_idx = x.size - 1
+            if y1_idx >= y.size:
+                y1_idx = y.size - 1
+            if z1_idx >= z.size:
+                z1_idx = z.size - 1
+
+            # Compute interpolation weights
+            wx = (xpi - x[x0_idx]) / diffs[0]
+            wy = (ypi - y[y0_idx]) / diffs[1]
+            wz = (zpi - z[z0_idx]) / diffs[2]
+
+            # Extract values values at corner points
+            c000 = values[x0_idx, y0_idx, z0_idx]
+            c001 = values[x0_idx, y0_idx, z1_idx]
+            c010 = values[x0_idx, y1_idx, z0_idx]
+            c011 = values[x0_idx, y1_idx, z1_idx]
+            c100 = values[x1_idx, y0_idx, z0_idx]
+            c101 = values[x1_idx, y0_idx, z1_idx]
+            c110 = values[x1_idx, y1_idx, z0_idx]
+            c111 = values[x1_idx, y1_idx, z1_idx]
+
+            # Perform trilinear interpolation
+            c00 = c000 * (1 - wx) + c100 * wx
+            c01 = c001 * (1 - wx) + c101 * wx
+            c10 = c010 * (1 - wx) + c110 * wx
+            c11 = c011 * (1 - wx) + c111 * wx
+
+            c0 = c00 * (1 - wy) + c10 * wy
+            c1 = c01 * (1 - wy) + c11 * wy
+
+            values_interp[i] = c0 * (1 - wz) + c1 * wz
+
+        return values_interp
+
+    return _interp3d
+
+
+def interp3d(axes_known, values, points_interp, extrap_fill_value=None):
+    _interp3d = _get_interp3d()
+    if extrap_fill_value is None:
+        extrap_fill_value = np.nan
+    return _interp3d(
+        axes_known=axes_known,
+        values=values,
+        points_interp=points_interp,
+        extrap_fill_value=extrap_fill_value,
+    )
+
+
+def interp_scipy(
+    axes_known,
+    values,
+    axes_interp: Sequence["np.ndarray"] = None,
+    points_interp: "np.ndarray" = None,
+    keep_dims=False,
+    bounds_error=True,
+    extrap_fill_value=None,
```

Comment:
@SimonBiggs, I had a similar issue where the clean test complained about missing `numpy` due to `np.nan` default parameter values. Is using None and setting in the function a reasonable workaround?

Reply:
Yup, that's perfect